README.md

```diff
@@ -31,6 +31,130 @@ Application-level tracing for Rust.
 structured, event-based diagnostic information. `tracing` is maintained by the
 Tokio project, but does _not_ require the `tokio` runtime to be used.
 
+## Usage
+
+### In libraries
+
+Libraries should only rely on the `tracing` crate and use the provided macros
+and types to collect whatever information might be useful to downstream consumers.
+
+```toml
+[dependencies]
+tracing = "0.1"
+```
+
+```rust
+use std::{error::Error, io};
+use tracing::{debug, error, info, span, warn, Level, instrument};
+
+#[instrument]
+pub fn shave(yak: usize) -> Result<(), Box<dyn Error + 'static>> {
+    debug!(excitement = "yay!", "hello! I'm gonna shave a yak.");
```

Comment:
it would be nice to at least briefly mention that `excitement = "yay"` is a structured key-value field with name "excitement" and value "yay", and that "hello i'm gonna shave a yak" is a human-readable `format_args!` message?